etesync-dav: radicale[1317]: [1317/Thread-7 (process_request_thread)] [ERROR] An exception occurred during PROPFIND request on '/${username}/': Storage.discover() takes from 2 to 3 positional arguments but 5 were given

## Describe the bug

With `services.etesync-dav.enable = true;` and a user account with `${username}` set up, and KDE PIM set up to sync with etesync-dav, I see errors in the etesync-dav logs:
```
❯ journalctl -u etesync-dav -b | cat
Dez 02 23:06:41 ${hostname} systemd[1]: Started etesync-dav - A CalDAV and CardDAV adapter for EteSync.
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] Logging of backtrace is disabled in this loglevel
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] Logging of backtrace is disabled in this loglevel
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] Loaded default config
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] Loaded arguments
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] Starting Radicale
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] auth type is 'radicale.auth.htpasswd'
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] auth htpasswd encryption is 'radicale.auth.htpasswd_encryption.plain'
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] storage type is 'etesync_dav.radicale.storage'
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] rights type is 'etesync_dav.radicale.rights'
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] web type is 'etesync_dav.radicale.web'
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] hook type is 'radicale.hook.none'
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] permit delete of collection: True
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] permit overwrite of collection: True
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] Listening on '[127.0.0.1]:37358'
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] Listening on '[::1]:37358'
Dez 02 23:06:41 ${hostname} radicale[1317]: [1317] [INFO] Radicale server ready
Dez 02 23:07:58 ${hostname} radicale[1317]: [1317/Thread-1 (process_request_thread)] [INFO] PROPFIND request for '/${username}/' with depth '0' received from ::1 using 'Mozilla/5.0 (X11; Linux x86_64) KIO/6.8 kioworker/6.8.0'
Dez 02 23:07:58 ${hostname} radicale[1317]: [1317/Thread-1 (process_request_thread)] [INFO] Access to '/${username}/' denied for anonymous user
Dez 02 23:07:58 ${hostname} radicale[1317]: [1317/Thread-1 (process_request_thread)] [INFO] PROPFIND response status for '/${username}/' with depth '0' in 0.000 seconds: 401 Unauthorized
Dez 02 23:07:58 ${hostname} radicale[1317]: [1317/Thread-2 (process_request_thread)] [INFO] PROPFIND request for '/${username}/' with depth '0' received from ::1 using 'Mozilla/5.0 (X11; Linux x86_64) KIO/6.8 kioworker/6.8.0'
Dez 02 23:07:58 ${hostname} radicale[1317]: [1317/Thread-2 (process_request_thread)] [INFO] Successful login: '${username}'
Dez 02 23:07:59 ${hostname} radicale[1317]: [1317/Thread-2 (process_request_thread)] [ERROR] An exception occurred during PROPFIND request on '/${username}/': Storage.discover() takes from 2 to 3 positional arguments but 5 were given
Dez 02 23:07:59 ${hostname} radicale[1317]: [1317/Thread-4 (process_request_thread)] [INFO] PROPFIND request for '/${username}/' with depth '0' received from ::1 using 'Mozilla/5.0 (X11; Linux x86_64) KIO/6.8 kioworker/6.8.0'
Dez 02 23:07:59 ${hostname} radicale[1317]: [1317/Thread-4 (process_request_thread)] [INFO] Access to '/${username}/' denied for anonymous user
Dez 02 23:07:59 ${hostname} radicale[1317]: [1317/Thread-4 (process_request_thread)] [INFO] PROPFIND response status for '/${username}/' with depth '0' in 0.000 seconds: 401 Unauthorized
Dez 02 23:07:59 ${hostname} radicale[1317]: [1317/Thread-5 (process_request_thread)] [INFO] PROPFIND request for '/${username}/' with depth '0' received from ::1 using 'Mozilla/5.0 (X11; Linux x86_64) KIO/6.8 kioworker/6.8.0'
Dez 02 23:07:59 ${hostname} radicale[1317]: [1317/Thread-5 (process_request_thread)] [INFO] Successful login: '${username}'
Dez 02 23:07:59 ${hostname} radicale[1317]: [1317/Thread-5 (process_request_thread)] [ERROR] An exception occurred during PROPFIND request on '/${username}/': Storage.discover() takes from 2 to 3 positional arguments but 5 were given
Dez 02 23:11:50 ${hostname} radicale[1317]: [1317/Thread-6 (process_request_thread)] [INFO] PROPFIND request for '/${username}/' with depth '0' received from ::1 using 'Mozilla/5.0 (X11; Linux x86_64) KIO/6.8 kioworker/6.8.0'
Dez 02 23:11:50 ${hostname} radicale[1317]: [1317/Thread-6 (process_request_thread)] [INFO] Access to '/${username}/' denied for anonymous user
Dez 02 23:11:50 ${hostname} radicale[1317]: [1317/Thread-6 (process_request_thread)] [INFO] PROPFIND response status for '/${username}/' with depth '0' in 0.000 seconds: 401 Unauthorized
Dez 02 23:11:50 ${hostname} radicale[1317]: [1317/Thread-7 (process_request_thread)] [INFO] PROPFIND request for '/${username}/' with depth '0' received from ::1 using 'Mozilla/5.0 (X11; Linux x86_64) KIO/6.8 kioworker/6.8.0'
Dez 02 23:11:50 ${hostname} radicale[1317]: [1317/Thread-7 (process_request_thread)] [INFO] Successful login: '${username}'
Dez 02 23:11:51 ${hostname} radicale[1317]: [1317/Thread-7 (process_request_thread)] [ERROR] An exception occurred during PROPFIND request on '/${username}/': Storage.discover() takes from 2 to 3 positional arguments but 5 were given
Dez 02 23:11:51 ${hostname} radicale[1317]: [1317/Thread-8 (process_request_thread)] [INFO] PROPFIND request for '/${username}/${some_id_like_thing}/' with depth '1' received from ::1 using 'Mozilla/5.0 (X11; Linux x86_64) KIO/6.8 kioworker/6.8.0'
Dez 02 23:11:51 ${hostname} radicale[1317]: [1317/Thread-8 (process_request_thread)] [INFO] Access to '/${username}/${some_id_like_thing}/' denied for anonymous user
Dez 02 23:11:51 ${hostname} radicale[1317]: [1317/Thread-8 (process_request_thread)] [INFO] PROPFIND response status for '/${username}/${some_id_like_thing}/' with depth '1' in 0.000 seconds: 401 Unauthorized
Dez 02 23:11:51 ${hostname} radicale[1317]: [1317/Thread-9 (process_request_thread)] [INFO] PROPFIND request for '/${username}/${some_id_like_thing}/' with depth '1' received from ::1 using 'Mozilla/5.0 (X11; Linux x86_64) KIO/6.8 kioworker/6.8.0'
Dez 02 23:11:51 ${hostname} radicale[1317]: [1317/Thread-9 (process_request_thread)] [INFO] Successful login: '${username}'
Dez 02 23:11:51 ${hostname} radicale[1317]: [1317/Thread-9 (process_request_thread)] [ERROR] An exception occurred during PROPFIND request on '/${username}/${some_id_like_thing}/': Storage.discover() takes from 2 to 3 positional arguments but 5 were given
Dez 02 23:16:50 ${hostname} radicale[1317]: [1317/Thread-10 (process_request_thread)] [INFO] PROPFIND request for '/${username}/' with depth '0' received from ::1 using 'Mozilla/5.0 (X11; Linux x86_64) KIO/6.8 kioworker/6.8.0'
Dez 02 23:16:50 ${hostname} radicale[1317]: [1317/Thread-10 (process_request_thread)] [INFO] Access to '/${username}/' denied for anonymous user
Dez 02 23:16:50 ${hostname} radicale[1317]: [1317/Thread-10 (process_request_thread)] [INFO] PROPFIND response status for '/${username}/' with depth '0' in 0.000 seconds: 401 Unauthorized
Dez 02 23:16:50 ${hostname} radicale[1317]: [1317/Thread-11 (process_request_thread)] [INFO] PROPFIND request for '/${username}/' with depth '0' received from ::1 using 'Mozilla/5.0 (X11; Linux x86_64) KIO/6.8 kioworker/6.8.0'
Dez 02 23:16:50 ${hostname} radicale[1317]: [1317/Thread-11 (process_request_thread)] [INFO] Successful login: '${username}'
Dez 02 23:16:51 ${hostname} radicale[1317]: [1317/Thread-11 (process_request_thread)] [ERROR] An exception occurred during PROPFIND request on '/${username}/': Storage.discover() takes from 2 to 3 positional arguments but 5 were given
Dez 02 23:16:51 ${hostname} radicale[1317]: [1317/Thread-12 (process_request_thread)] [INFO] PROPFIND request for '/${username}/${some_id_like_thing}/' with depth '1' received from ::1 using 'Mozilla/5.0 (X11; Linux x86_64) KIO/6.8 kioworker/6.8.0'
Dez 02 23:16:51 ${hostname} radicale[1317]: [1317/Thread-12 (process_request_thread)] [INFO] Access to '/${username}/${some_id_like_thing}/' denied for anonymous user
Dez 02 23:16:51 ${hostname} radicale[1317]: [1317/Thread-12 (process_request_thread)] [INFO] PROPFIND response status for '/${username}/${some_id_like_thing}/' with depth '1' in 0.000 seconds: 401 Unauthorized
Dez 02 23:16:51 ${hostname} radicale[1317]: [1317/Thread-13 (process_request_thread)] [INFO] PROPFIND request for '/${username}/${some_id_like_thing}/' with depth '1' received from ::1 using 'Mozilla/5.0 (X11; Linux x86_64) KIO/6.8 kioworker/6.8.0'
Dez 02 23:16:51 ${hostname} radicale[1317]: [1317/Thread-13 (process_request_thread)] [INFO] Successful login: '${username}'
Dez 02 23:16:51 ${hostname} radicale[1317]: [1317/Thread-13 (process_request_thread)] [ERROR] An exception occurred during PROPFIND request on '/${username}/${some_id_like_thing}/': Storage.discover() takes from 2 to 3 positional arguments but 5 were given
```

Are we running etesync-dav with a version of Radicale it does not support?

```
❯ ls /nix/store/*radicale*
 #                                name                                type    size       modified
 0   /nix/store/g78l864aqv19dxzagx16v5f7hdhakkil-radicale-3.3.1       dir       34 B   54 years ago
 1   /nix/store/mjx00szdhhfj624kz61x91r4qvjgr0mr-radicale-3.3.1.drv   file   5.0 KiB   54 years ago
```

https://github.com/etesync/etesync-dav/releases/tag/v0.33.4 has the release note:

> Downgrade Radicale version requirements to 3.2.0 (there are issues with 3.3.0).

See-also: https://github.com/NixOS/nixpkgs/issues/361242

## Expected behavior

<!-- A clear and concise description of what you expected to happen. -->

etesync-dav runs without any error logs.

## Screenshots

<!-- If applicable, add screenshots to help explain your problem: -->

## Additional context

<!-- Add any other context about the problem here. -->
```
❯ systemctl status etesync-dav
● etesync-dav.service - etesync-dav - A CalDAV and CardDAV adapter for EteSync
     Loaded: loaded (/etc/systemd/system/etesync-dav.service; enabled; preset: ignored)
     Active: active (running) since Mon 2024-12-02 23:06:41 CET; 17min ago
 Invocation: 2f1a3414e9e94429950b36fcc503bf65
   Main PID: 1317 (.etesync-dav-wr)
         IP: 49.6K in, 31.8K out
         IO: 74.9M read, 56.1M written
      Tasks: 3 (limit: 37376)
     Memory: 98.4M (peak: 99.4M)
        CPU: 504ms
     CGroup: /system.slice/etesync-dav.service
             └─1317 /nix/store/px2nj16i5gc3d4mnw5l1nclfdxhry61p-python3-3.12.7/bin/python3.12 /nix/store/72r4hynzrdbf3fhjwprz93451cg6yq5s-etesync-dav-0.32.1-unstable-2024-09-02/bin/.etesync-dav-wrapped
[...]

❯ /nix/store/px2nj16i5gc3d4mnw5l1nclfdxhry61p-python3-3.12.7/bin/python3.12 /nix/store/72r4hynzrdbf3fhjwprz93451cg6yq5s-etesync-dav-0.32.1-unstable-2024-09-02/bin/.etesync-dav-wrapped --version
EteSync DAV version:  0.32.1
Radicale version:  3.3.1
```

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

```
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.1, NixOS, 25.05 (Warbler), 25.05.20241128.970e93b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - channels(root): `"nixos"`
 - channels(${USER}): `""`
 - nixpkgs: `/nix/store/h5v6brw3yj14b52a58imzjm3mid6scsz-source`
```

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@thyol, @Valodim

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/et/etesync-dav/package.nix
+++ b/pkgs/by-name/et/etesync-dav/package.nix
@@ -8,14 +8,14 @@
 }:
 python3Packages.buildPythonApplication rec {
   pname = "etesync-dav";
-  version = "0.34.0";
+  version = "0.35.0";
   format = "pyproject";
 
   src = fetchFromGitHub {
     owner = "etesync";
     repo = "etesync-dav";
     tag = "v${version}";
-    hash = "sha256-+rNqyksOmDUh0OuvgEDWv6tuZQkn1gizz35Ptr6izos=";
+    hash = "sha256-CD02nuA9GD/oe7mjExUHIftkPxM1pZQKyDalXSoOhXY=";
   };
 
   build-system = with python3Packages; [ setuptools ];

--- a/pkgs/by-name/et/etesync-dav/package.nix
+++ b/pkgs/by-name/et/etesync-dav/package.nix
@@ -9,7 +9,7 @@
 python3Packages.buildPythonApplication rec {
   pname = "etesync-dav";
   version = "0.35.0";
-  format = "pyproject";
+  pyproject = true;
 
   src = fetchFromGitHub {
     owner = "etesync";

